primus.js

```diff
@@ -416,12 +410,11 @@ Primus.prototype.initialise = function initialise(options) {
     primus.emit('open');
   });
 
-  primus.on('incoming::pong', function pong(time) {
+  primus.on('incoming::ping', function ping(time) {
     primus.online = true;
-    primus.timers.clear('pong');
     primus.heartbeat();
-
-    primus.latency = (+new Date()) - time;
+    primus.emit('outgoing::pong', time);
+    primus._write('primus::pong::'+ time);
```

Comment:
If for some odd reason this is called when `readyState !== Primus.OPEN` the message will be queued. It shouldn't be a big issue though as I _think_ that sending a "spurious" pong will not cause any harm.